tests/basic/clocks.rs

```diff
@@ -437,3 +440,37 @@ fn clock_fetch_update() {
         None,
     );
 }
+
+fn clock_once(num_threads: usize) {
+    let once = Arc::new(Once::new());
+    let init = Arc::new(std::sync::atomic::AtomicUsize::new(0));
+
+    let threads = (0..num_threads)
+        .map(|_| {
+            let once = Arc::clone(&once);
+            let init = Arc::clone(&init);
+            thread::spawn(move || {
+                check_clock(|i, c| (c > 0) == (i == 0));
+                once.call_once(|| init.store(me(), std::sync::atomic::Ordering::SeqCst));
+                let who_inited = init.load(std::sync::atomic::Ordering::SeqCst);
+                // should have inhaled the clock of the thread that inited the Once, but might also
+                // have inhaled the clocks of threads that we were racing with for initialization
+                check_clock(|i, c| i != who_inited || c > 0);
```

Comment:
I think this can be a stronger check.  You can assert that the clocks of the following threads are nonzero: {0, who_inited, me}, i.e.,
`check_clock(|i, c| !(i == who_inited || i == 0 || i == me()) || c>0)`